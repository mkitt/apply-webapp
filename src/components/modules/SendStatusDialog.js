// @flow
import React from 'react'
import Select from '../elements/Select'
import Dialog from '../blocks/Dialog'

type Props = {
  handleCancel: Function,
  handleChange: Function,
  handleConfirm: Function,
  statusValue: string,
}

export default(props: Props) => (
  <Dialog
    confirmLabel="Send status email"
    handleCancel={props.handleCancel}
    handleConfirm={props.handleConfirm}
    title="Send Status Email"
  >
    <Select
      label="Choose Status"
      name="recommendationSelect"
      onChange={props.handleChange}
      value={props.statusValue}
    >
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
      <option value="option4">Option 4</option>
      <option value="option5">Option 5</option>
    </Select>
  </Dialog>
)
