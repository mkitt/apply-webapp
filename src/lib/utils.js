// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
