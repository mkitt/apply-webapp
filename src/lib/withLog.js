import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
