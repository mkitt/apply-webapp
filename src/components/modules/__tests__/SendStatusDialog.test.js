import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from '../../../lib/utils'
import SendStatusDialog from '../SendStatusDialog'

describe('<SendStatusDialog />', () => {
  it('shows the correct snapshot tree for the <SendStatusDialog> module', () => {
    const component = (
      <SendStatusDialog
        handleCancel={noop}
        handleChange={noop}
        handleConfirm={noop}
        statusValue="Status Value"
      />
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
