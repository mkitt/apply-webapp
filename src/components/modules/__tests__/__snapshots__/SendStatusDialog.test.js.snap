// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendStatusDialog /> shows the correct snapshot tree for the <SendStatusDialog> module 1`] = `
.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 112px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-1,
[data-css-1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  font-size: 34px;
  font-weight: 200;
  color: #808080;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-2,
[data-css-2] {
  width: 100%;
  margin-top: 16px;
  margin-bottom: 32px;
  font-size: 14px;
  color: #808080;
}

.css-3,
[data-css-3] {
  position: relative;
}

.css-4,
[data-css-4] {
  -webkit-appearance: button;
  -webkit-border-radius: 0;
  -webkit-box-shadow: none;
  -webkit-padding-end: 0;
  -webkit-padding-start: 0;
  -webkit-user-select: none;
  width: 100%;
  height: 48px;
  margin: 0;
  padding: 0 32px;
  overflow: hidden;
  font-size: 14px;
  font-family: inherit;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
  background-image: none;
  background-position: initial;
  background-repeat: no-repeat;
  border-radius: 0;
  border: 0;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.css-4:focus,
[data-css-4]:focus {
  border-color: #999;
  outline: 0;
}

.css-6,
[data-css-6] {
  position: absolute;
  top: calc(50% - 8px);
  right: 32px;
  color: #fff;
  pointer-events: none;
}

.css-7,
[data-css-7] {
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 16px;
  -ms-grid-template-columns: 1fr;
  -ms-grid-gap: 16px;
}

.css-8,
[data-css-8] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #999;
}

.css-9,
[data-css-9] {
  text-align: left;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
}

@media (min-width: 31em) {
  .css-0,
  [data-css-0] {
    width: 496px;
  }
}

@media (min-width: 31em) {
  .css-2,
  [data-css-2] {
    margin-bottom: 64px;
  }
}

@media (min-width: 31em) {
  .css-7,
  [data-css-7] {
    grid-template-columns: 1fr 1fr;
    -ms-grid-template-columns: 1fr 1fr;
  }
}

<div
  className="css-0"
>
  <h2
    className="css-1"
  >
    Send Status Email
  </h2>
  <div
    className="css-2"
  >
    <label
      className="css-3"
      htmlFor="recommendationSelect"
    >
      <select
        aria-label="Choose Status"
        className="css-4"
        id="recommendationSelect"
        name="recommendationSelect"
        onChange={[Function]}
        value="Status Value"
      >
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
        <option
          value="option3"
        >
          Option 3
        </option>
        <option
          value="option4"
        >
          Option 4
        </option>
        <option
          value="option5"
        >
          Option 5
        </option>
      </select>
      <div
        className="css-6"
      >
        <svg
          className="css-nil"
          height={12}
          viewBox="0 0 21 12"
          width={21}
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          >
            <polyline
              points="19.182 1.067 10.611 10.067 1.182 1.067"
            />
          </g>
        </svg>
      </div>
    </label>
  </div>
  <div
    className="css-7"
  >
    <button
      className="css-8"
      onClick={[Function]}
      role="button"
    >
      cancel
    </button>
    <button
      className="css-9"
      onClick={[Function]}
      role="button"
    >
      Send status email
    </button>
  </div>
</div>
`;
